openfl/transport/grpc/aggregator_server.py

```diff
@@ -206,9 +209,102 @@ def SendLocalTaskResults(self, request, context):  # NOQA:N802
             header=self.get_header(collaborator_name)
         )
 
+    def GetMetricStream(self, request, context):  # NOQA:N802
+        """Get metric stream."""
+        while True:
+            try:
+                logger.debug('Try to get metric dict')
+                metric_dict = self.aggregator.metric_queue.get(timeout=1)
+            except queue.Empty:
+                if self.aggregator.all_quit_jobs_sent() and self.aggregator.metric_queue.empty():
+                    return
+                continue
+            logger.debug('Metric dict has got. Yield it')
+            yield aggregator_pb2.GetMetricStreamResponse(**metric_dict)
+
+    def GetTrainedModel(self, request, context):  # NOQA:N802
```

Comment:
If you are going to use the same gRPC server for director and collaborators, there should to be some way to filter between them to limit access only to intended APIs for each role. The three prior endpoints all used the helper function `validate_collaborator`, so a helper function like `validate_director` would be of use here

Reply:
Method `validate_director_request`. It checks that a subject name from client certificate matches an aggregator FQDN (Director and Aggregator FQDNs are same and we set director FQDN as a subject name for certificate https://openfl.readthedocs.io/en/latest/source/utilities/pki.html#semi-automatic-certification). Collaborator's and user's request will be rejected.